src/scripts/setup.js

```diff
@@ -57,7 +68,35 @@ function setup_delete() {
   });
 }
 
-export {
-  setup_modify,
-  setup_delete
-};
+export function setup_tag_recommend(search_tag) {
+  // set up listener for new_modal
+  const new_modal_tag = document.querySelector('#new-modal-tag');
+  new_modal_tag.addEventListener('input', () => {
+    handle_tag_input_change(search_tag, new_modal_tag, 'new');
+  });
+
+  // set up listener for info_modal
+  const info_modal_tag = document.querySelector('#info-modal-input-tag');
+  info_modal_tag.addEventListener('input', () => {
+    handle_tag_input_change(search_tag, info_modal_tag, 'info');
+  });
+}
+
+function handle_tag_input_change(search_tag, tag_input, type) {
```

Comment:
why is search_tag() passed through 3 functions instead of directly exporting from home.js?